### PR TITLE
90% PLAT-2382 Adds clearer persona errors.

### PR DESCRIPTION
**Pull request by bilaal-s**
This PR adds further error handling for the persona authorisation error responses. talis/platform#2382

### CODE REVIEW

File: echo/index.js [hunk 1]
```diff
@@ -124,13 +124,12 @@ function Client() {
...
-  function getCodeFromHttpStatusCode(err, httpStatusCode) {
+  function getCodeFromHttpStatusCode(err, httpStatusCode, responseBody) {
     if (err) {
       return codesAndLabels.REQUEST_ERROR;
-    } else if (httpStatusToCode[httpStatusCode]) {
-      return httpStatusToCode[httpStatusCode];
+    } else if (httpStatusToCode(httpStatusCode, responseBody)) {
+      return httpStatusToCode(httpStatusCode, responseBody);
     }
-    return codesAndLabels.UNKNOWN_ERROR;
```
Comment (markwallsgrove) [inline echo/index.js:133]: what happens if the last else if doesn't execute? Shouldn't something be returned?

File: echo/lib/codes-labels.js [hunk 2]
```diff
@@ -17,19 +17,49 @@ var codesAndLabels = {
...
 };
 
-var httpStatusToCode = {
-    304: codesAndLabels.NOT_MODIFIED,
-    400: codesAndLabels.INVALID_QUERY,
-    401: codesAndLabels.UNAUTHORIZED,
-    500: codesAndLabels.INTERNAL_ERROR,
-    501: codesAndLabels.NOT_IMPLEMENTED
+var httpStatusToCode = function(httpStatusCode, responseBody) {
+    var code;
```
Comment (markwallsgrove) [inline echo/lib/codes-labels.js]: this variable isn't needed. Just return

Comment (markwallsgrove) [reply echo/lib/codes-labels.js]: you can then remove all of the `break;` lines too

Comment (bilaal-s) [reply echo/lib/codes-labels.js]: Sure thing!

File: echo/index.js (hunk 1 above)
Comment (bilaal-s) [reply echo/index.js:133]: Yes you are right, I will add the unknown error back here

**Review by markwallsgrove — APPROVED** [review]

**Pull request merged**